File: fix_files.py
import os
import re

base_dir = r"c:\Users\ajst1\Phantom\PhantomFrame"

for root, _, files in os.walk(base_dir):
    for f in files:
        if f.endswith(".js") or f.endswith(".css"):
            path = os.path.join(root, f)
            with open(path, "r", encoding="utf-8") as file:
                content = file.read()
            
            # Simple heuristic: if it looks like python script wrapping JS/CSS
            if '"""' in content or '"" "' in content:
                start = content.find('"""')
                if start == -1:
                    start = content.find('"" "')
                    if start != -1:
                        start += 4
                else:
                    start += 3
                
                end = content.rfind('"""')
                if end == -1 or end < start:
                    end = content.rfind('"" "')
                
                if start != -1 and end != -1 and start < end:
                    # check if the part before start contains "with open" or "= \"\"\""
                    prefix = content[:start]
                    if "=" in prefix or "with open" in content:
                        actual_code = content[start:end]
                        # Remove leading newline if present
                        if actual_code.startswith('\n'):
                            actual_code = actual_code[1:]
                        with open(path, "w", encoding="utf-8") as file:
                            file.write(actual_code)
                        print(f"Fixed {path}")

File: fix_files.js
const fs = require('fs');
const path = require('path');

const baseDir = path.join('c:', 'Users', 'ajst1', 'Phantom', 'PhantomFrame');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(filePath));
        } else {
            if (filePath.endsWith('.js') || filePath.endsWith('.css')) {
                results.push(filePath);
            }
        }
    });
    return results;
}

const files = walk(baseDir);

files.forEach(file => {
    const content = fs.readFileSync(file, 'utf8');
    let start = -1;
    let end = -1;
    
    if (content.includes('"""')) {
        start = content.indexOf('"""');
        end = content.lastIndexOf('"""');
        if (start !== end && start !== -1) {
            start += 3;
        }
    } else if (content.includes('"" "')) {
        start = content.indexOf('"" "');
        end = content.lastIndexOf('"" "');
        if (start !== end && start !== -1) {
            start += 4;
        }
    }
    
    if (start !== -1 && end !== -1 && start < end) {
        let prefix = content.substring(0, start - 3);
        if (prefix.includes('=') || prefix.includes('with open')) {
            let actualCode = content.substring(start, end);
            if (actualCode.startsWith('\n')) {
                actualCode = actualCode.substring(1);
            }
            if (actualCode.startsWith('\r\n')) {
                actualCode = actualCode.substring(2);
            }
            fs.writeFileSync(file, actualCode, 'utf8');
            console.log(`Fixed ${file}`);
        }
    }
});
